### PR TITLE
Introduce client_id in SensorMetadata

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/cloud/internal/map_builder_server.h [hunk 1]
```diff
@@ -110,8 +110,8 @@ class MapBuilderServer : public MapBuilderServerInterface {
   void ProcessSensorDataQueue();
   void StartSlamThread();
   void OnLocalSlamResult(
-      int trajectory_id, common::Time time, transform::Rigid3d local_pose,
-      sensor::RangeData range_data,
+      int trajectory_id, const std::string& client_id, common::Time time,
```
Comment (gaschler) [inline cartographer/cloud/internal/map_builder_server.h]: the other arguments are passed by value, also pass client_id by value?

File: cartographer/cloud/internal/map_builder_context_impl.h [hunk 2]
```diff
@@ -40,14 +40,19 @@ template <class SubmapType>
 mapping::TrajectoryBuilderInterface::LocalSlamResultCallback
 MapBuilderContext<SubmapType>::GetLocalSlamResultCallbackForSubscriptions() {
   MapBuilderServer* map_builder_server = map_builder_server_;
```
Comment (gaschler) [inline cartographer/cloud/internal/map_builder_context_impl.h]: you can probably drop this line when you capture `this` anyways

File: cartographer/cloud/internal/map_builder_server.h (hunk 1 above)
Comment (cschuet) [reply cartographer/cloud/internal/map_builder_server.h]: done.

File: cartographer/cloud/internal/map_builder_context_impl.h (hunk 2 above)
Comment (cschuet) [reply cartographer/cloud/internal/map_builder_context_impl.h]: done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**